### PR TITLE
Add top banner of home page and join us section

### PR DESCRIPTION
**Pull request by michaelkli1998**
Added first two sections of home page and made it mobile responsive!
Screenshot 2023-07-12 at 4 22 09 PM

### CODE REVIEW

File: frontend/src/components/Button.js
Comment (jchenggggg) [inline frontend/src/components/Button.js:4]: Dang it you beat me to it

File: frontend/src/pages/index.js
Comment (jchenggggg) [inline frontend/src/pages/index.js:3]: Were you planning on using these?

File: frontend/src/components/Button.js
Comment (jchenggggg) [inline frontend/src/components/Button.js:6]: I feel like it's more clear either to make a component specifically called `PrimaryButtonLink` or `SecondaryButtonLink` (as I did in my PR), or to pass in the string `'primary'` or `'secondary'` explicitly to specify the style, instead of having to know that `primary = false` makes the button look a certain way. But it's not a big deal, whatever works

**Review by jchenggggg — COMMENTED** [review]
I don't know css stuff so I'll let @thedomcab approve. Looks good tho

File: frontend/src/components/Button.js
Comment (thedomcab) [reply frontend/src/components/Button.js:6]: I agree with what @jchenggggg  has because we might have more customization of buttonlink in the future.  once @jchenggggg merge his into main you can rebase and use his logic.

File: frontend/src/pages/index.js
Comment (thedomcab) [inline frontend/src/pages/index.js:18]:
```suggestion
          <div className="flex md:flex-row flex-col justify-between items-center py-[50px] max-w-container w-full gap-4">
```
have the same tailwind tags for several breakpoints is redundant. pick the minimum breakpoint and it should stick.   like i suggested.

File: frontend/src/pages/index.js
Comment (thedomcab) [inline frontend/src/pages/index.js:19]:
```suggestion
            <h2 className="mb-0 xl:text-[60px] lg:text-[55px] md:text-[50px] sm:text-[45px] text-[40px]  sm:text-left text-center font-bold leading-tight tracking-[3.6px]">JOIN US THIS <br/> SUNDAY!</h2>
```
same thing here.

File: frontend/src/pages/index.js
Comment (thedomcab) [inline frontend/src/pages/index.js:20]:
```suggestion
            <div className="flex flex-col items-center  md:items-start">
```
ditto.

File: frontend/src/pages/index.js
Comment (thedomcab) [inline frontend/src/pages/index.js:23]:
```suggestion
              <div className="flex sm:justify-start justify-center text-center ">
```
ditto.

File: frontend/src/pages/index.js
Comment (michaelkli1998) [reply frontend/src/pages/index.js:18]: Oh awesome, didn't know that, will make these changes!

File: frontend/src/components/Button.js
Comment (michaelkli1998) [reply frontend/src/components/Button.js:6]: Will do!

Comment (jchenggggg) [reply frontend/src/components/Button.js:6]: I merged #28 !

**Review by thedomcab — APPROVED** [review]

**Pull request merged**
